### PR TITLE
[TA-2683] Setup tests and add tests for listing assignments

### PR DESCRIPTION
**Pull request by LaberionAjvazi**
#### Description

Setup testing with jest and add tests for list assignments command.

#### Checklist

- [x] I have self-reviewed this PR
- [x] I have tested the change and proved that it works in different scenarios
- [ ] I have updated docs if needed

### CODE REVIEW

File: jest.config.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import type { Config } from "@jest/types"
+const config: Config.InitialOptions = {
+    verbose: true,
+    transform: {
+        "^.+\\.tsx?$": "ts-jest",
```
Comment (promeris) [inline jest.config.ts:5]: Do we need this configuration as there aren't any tsx files?

File: tests/utls/context-mock.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import {contextService} from "../../src/services/context.service";
+
+export function setDefaultProfile(): void {
+    contextService.setContext({
+        profile: {
+            name: "test",
+            team: "https://myTeam.celonis.cloud/",
+            apiToken: "YnQ3N2M0M2ItYzQ3OS00YzgyLTg0ODgtOWNkNzhiNzYwOTU2OlFkNnBpVCs0M0JBYm1ZWGlCZ2hPd245aldwWTNubFQyYVFOTFBUeHEwdUxM",
```
Comment (promeris) [inline tests/utls/context-mock.ts:8]: I don't think we need a real token here :smile:.

File: tests/utls/context-mock.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import {contextService} from "../../src/services/context.service";
+
+export function setDefaultProfile(): void {
```
Comment (qmucolli) [inline tests/utls/context-mock.ts:3]: Is this needed on a global level (should all tests have this by default)?

File: tests/utls/test-transport.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import * as Transport from "winston-transport";
+import {LogEntry} from "winston";
+
+export class TestTransport extends Transport {
```
Comment (qmucolli) [inline tests/utls/test-transport.ts:4]: Could you elaborate what this is needed for? I'm not sure I'm following :D

File: tests/list/assignments.spec.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
+import {VariableCommand} from "../../src/commands/variable.command";
+import axios from "axios";
+import {logger} from "../../src/util/logger";
+import {TestTransport} from "../utls/test-transport";
+import {setDefaultProfile} from "../utls/context-mock";
+import {FileService} from "../../src/services/file-service";
+import * as fs from "fs";
+import * as path from "path";
+
+jest.mock("axios");
```
Comment (qmucolli) [inline tests/list/assignments.spec.ts:10]: I think these are better suited as part of a global setup instead of here, wdyt?

**Review by qmucolli — COMMENTED** [review]
Generally looks good, I'm wondering if we should think of adding SonarCloud and also handle a global test config which would do the initial generation of each test?

File: jest.config.ts (hunk 1 above)
Comment (LaberionAjvazi) [reply jest.config.ts:5]: `x` is optional in this expression. We need this to transform our `ts` files before running the tests.

File: tests/utls/context-mock.ts (hunk 2 above)
Comment (LaberionAjvazi) [reply tests/utls/context-mock.ts:8]: This is a non-valid mocked token anyway.

File: tests/utls/context-mock.ts (hunk 3 above)
Comment (LaberionAjvazi) [reply tests/utls/context-mock.ts:3]: Apart from tests for profile commands, I think all tests will need this.

File: tests/utls/test-transport.ts (hunk 4 above)
Comment (LaberionAjvazi) [reply tests/utls/test-transport.ts:4]: I am using this to capture logs during testing. By attaching a custom transport (similar to [this](https://github.com/celonis/content-cli/blob/master/src/util/logger.ts#L24)) I can then read the logs as you would e.g. in Java by adding a TestAppender to logback.

File: tests/list/assignments.spec.ts (hunk 5 above)
Comment (LaberionAjvazi) [reply tests/list/assignments.spec.ts]: I agree and added a test setup file that handles mocking modules and the default profile.

**Review by promeris — APPROVED** [review]

**Review by promeris — DISMISSED** [review]

**Review by qmucolli — DISMISSED** [review]

**Review by promeris — DISMISSED** [review]

**Review by promeris — APPROVED** [review]

**Pull request merged**